Assets/Scripts/Model/GameManager.cs

```diff
@@ -203,35 +199,30 @@ public void cardDraggedToFrontCommand(Position playerPosition, int cardId)
             playerSouth.armymodel.armyCardsModel.moveCardFromHandToFront(cardId);
         }
     }
-
-    public void attributeNamesAndFactions()
-    {
-        //Factions
-        firstFaction = chooseFactionForFirstPlayer.getFirstFaction();
-        secondFaction = chooseFactionForSecondPlayer.getSecondFaction();
-
-        //Names
-        southName = chooseFactionForFirstPlayer.getFirstPlayersName();
-        northName = chooseFactionForSecondPlayer.getSecondPlayersName();
-        if (string.IsNullOrEmpty(southName)) southName = "Gracz 1";
-        if (string.IsNullOrEmpty(northName)) northName = "Gracz 2";
-    }
-
     public void StartGameWithCouroutine()
     {
         StartCoroutine(startGame());
     }
 
-    public void drawNewCardPlayerNorth()
+    public void drawNewCard(PlayerModel playerModel, HandView handView, GameObject deck, bool isCoroutine)
     {
-        Card cardDrawn = playerNorth.armymodel.armyCardsModel.moveCardFromDeckListToHandList();
-        northHandView.MoveDrawnCardFromDeckToHand(cardDrawn, playerNorth, deckNorth);
-    }
+        if (isCoroutine)
```

Comment:
please find a better name for is Coroutine - variable does not tell much - is it coroutine in general or any specific?